src/videopython/base/video.py

```diff
@@ -104,16 +142,90 @@ def __init__(self):
     @classmethod
     def from_path(cls, path: str) -> Video:
         new_vid = cls()
-        new_vid.frames, new_vid.fps = cls._load_video_from_path(path)
 
         try:
-            new_vid.audio = Audio.from_file(path)
-        except Exception as e:
-            print(f"No audio found for `{path}`, adding silent track!")
-            new_vid.audio = Audio.create_silent(
-                duration_seconds=round(new_vid.total_seconds, 2), stereo=True, sample_rate=44100
+            # Get video metadata using VideoMetadata.from_path
+            metadata = VideoMetadata.from_path(path)
+
+            width = metadata.width
+            height = metadata.height
+            fps = metadata.fps
+            total_frames = metadata.frame_count
+
+            # Set up FFmpeg command for raw video extraction
+            ffmpeg_cmd = [
+                "ffmpeg",
+                "-i",
+                path,
+                "-f",
+                "rawvideo",
+                "-pix_fmt",
+                "rgb24",
+                "-vsync",
+                "0",
+                "-vcodec",
+                "rawvideo",
+                "-y",
+                "pipe:1",
+            ]
+
+            # Start FFmpeg process
+            process = subprocess.Popen(
+                ffmpeg_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, bufsize=10**8  # Use large buffer
             )
-        return new_vid
+
+            # Calculate frame size in bytes
+            frame_size = width * height * 3  # 3 bytes per pixel for RGB
+
+            # Pre-allocate numpy array for all frames
+            frames = np.empty((total_frames, height, width, 3), dtype=np.uint8)
+
+            # Read frames in batches
+            BATCH_SIZE = 100
```

Comment:
Maybe `BATCH_SIZE` should be an argument instead of constant?

Reply:
Added 😎